### PR TITLE
refactor: Remove `throwOnPluginPubspecError` flag for plugin validation

### PR DESCRIPTION
**Pull request by Gustl22**
Part of #137040 and #80374

The flag `throwOnPluginPubspecError` never actually was enabled during production in #79669, but only in some dart plugin tests. And in the tests the case of the error when enabling the flag was not explicitly tested. The only thing tested was, that it is not thrown when disabled.

As explained [here](https://github.com/flutter/flutter/pull/142035#discussion_r1484237904) the only case, where this error could be thrown is, when a dart implementation and a native inline implementation are provided simultaneously. But throwing an exception there is a wrong behavior, as both can coexist in a plugin package, thus in the pubspec file.

Disabling the flag means, that the error is not thrown and not shown to the user. This is the case in production (contrary to the dart plugin tests), which acts like these plugin cases of implementations are just skipped. And this is what actually should be done.

In conclusion, I think the case of coexisting dart and native implementation in pubspec was just overlooked and therefore this error validation was introduced, which is not necessary or even valid.

For more discussion, see: https://discord.com/channels/608014603317936148/608022056616853515/1200194937791205436

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
  - This is tricky: I already added a test in #142035, which finally complies with the other tests, by removing the flag. So I think it falls in the category of "remove dead code".
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
  - Theoretically this is a breaking change, as removing / altering some tests. But the flag actually was never valid or used, so IDK. But this may not does fall in the category of "contributed tests".
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/dart_plugin_test.dart [hunk 1]
```diff
@@ -139,7 +139,6 @@ void main() {
               appDependencies: directDependencies,
             ),
           ],
-          throwOnPluginPubspecError: false,
```
Comment (Gustl22) [inline packages/flutter_tools/test/general.shard/dart_plugin_test.dart:142]: Note that this test failed before, when setting `throwOnPluginPubspecError: true`, which was a wrong behavior.

**Review by stuartmorgan-g — APPROVED** [review]
LGTM. My recollection is that this was disabled pretty much immediately because the logic wasn't actually right.

I think the only case I've seen in recent memory where an error message would have been useful is the case where a desktop plugin doesn't get autoregistration because of a lack of `implements`, but even that is unlikely to be useful going forward since new plugins are unlikely to support older than Flutter 2.11.

Given that, just removing it as cruft seems like the right choice.

**Comment by Gustl22** [comment]
I appreciate your speedy review and taking the time!

**Review by christopherfujino — APPROVED** [review]
(un)-ship it!

**Pull request merged**